pkg/cloud-controller-manager/instance.go

```diff
@@ -48,11 +50,23 @@ func (i *instanceManager) InstanceShutdown(ctx context.Context, node *v1.Node) (
 }
 
 func (i *instanceManager) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprovider.InstanceMetadata, error) {
-	vm, err := i.vmClient.Get(i.namespace, node.Name, metav1.GetOptions{})
+	vmi, err := i.vmiClient.Get(i.namespace, node.Name, metav1.GetOptions{})
```

Comment:
if  vm exist but vmi not exist, return a metadata which have `ProviderID`.
if vm exist and vmi  exist, return a metadata which have `ProviderID` `Zone` `Region`
if vm not exist, return error